sstables/compaction.cc

```diff
@@ -744,8 +744,8 @@ class regular_compaction : public compaction {
     }
 
     virtual flat_mutation_reader::filter make_partition_filter() const override {
-        return [] (const dht::decorated_key& dk){
-            return dht::shard_of(dk.token()) == engine().cpu_id();
+        return [s = _schema] (const dht::decorated_key& dk){
```

Comment:
`&s = *_schema` eliminates some reference-count bookkeeping. The reader should hole _schema for us.

Reply:
Is it guaranteed that a flat_mutation_reader will live longer than this lambda?

Reply:
The `_schema` member is from the `compaction` class, and that will outlive the reader.

Reply:
nvm. It's schema of sstable.

Reply:
done.